incubator/hnc/test/conversion/conversion_test.go

```diff
@@ -0,0 +1,144 @@
+		CheckGoPath()
+		CleanupNamespaces(nsA, nsB, nsC)
+		// Tear down HNC for both the specified version and at HEAD.
+		TearDownHNC(hncVersion)
+		// Wait for HNC to be fully torn down.
+		ensureHNCCleared(crds)
+	})
+
+	AfterEach(func() {
+		CleanupNamespaces(nsA, nsB, nsC)
+		// Only tear down HNC at HEAD, since that's what we just deployed.
+		TearDownHNC("")
+	})
+
+	It("should convert CRD versions properly and HNC should still work", func() {
+		// Set up v1alpha1 and verify everything is fine before conversion.
+		GinkgoT().Log("Set up v1alpha1")
+		setupV1alpha1(hncVersion)
+		// Wait for the validating webhooks to be ready.
+		ensureVWHReady()
+		// Create namespace 'a' and a subnamespace 'b'
+		GinkgoT().Log("Apply v1alpha1 resources")
+		MustRun("kubectl apply -f v1alpha1_resources.yaml")
+		// Create a namespace 'c' and use kubectl plugin to set its parent to 'a'
+		MustRun("kubectl create ns", nsC)
+		GinkgoT().Log("Use v1alpha1 kubectl-hns plugin")
+		MustRun("kubectl hns set", nsC, "--parent", nsA)
+
+		// Verify kubectl-hns plugin works and 'b' and 'c' are children of 'a'.
+		RunShouldContainMultiple([]string{nsB, nsC}, 2, "kubectl hns tree", nsA)
+		// Verify there's no 'v1alpha2' in all three CRDs for now.
+		checkCRDVersionInField("v1alpha2", ".spec", crds, false,2)
+		checkCRDVersionInField("v1alpha2", ".status", crds, false,2)
+
+		// Deploy the HEAD to include API conversion and install v1alpha2 kubectl plugin.
+		GinkgoT().Log("Set up v1alpha2")
+		setupV1alpha2()
+		// Wait for the cert rotator to write caBundles in CRD conversion webhooks.
+		ensureCRDConvWHReady(crds)
+
+		// Verify the conversion works with 'v1alpha1' still in the spec.versions but not in status.storedVersions.
+		checkCRDVersionInField("v1alpha1", ".spec.versions", crds, true, 2)
+		checkCRDVersionInField("v1alpha2", ".spec.versions", crds, true, 2)
+		// Give this a longer timeout for the reconciler to remove v1alpha1 from storedVersions.
+		checkCRDVersionInField("v1alpha1", ".status.storedVersions", crds, false, 5)
+		checkCRDVersionInField("v1alpha2", ".status.storedVersions", crds, true,2)
+
+		// Verify v1alpha2 kubectl-hns plugin still works
+		RunShouldContainMultiple([]string{nsB, nsC}, 2, "kubectl hns tree", nsA)
+		// Verify cascading deletion still works.
+		MustRun("kubectl hns set", nsA, "-a")
+		MustRun("kubectl delete ns", nsA)
+		// Verify conditions still work.
+		RunShouldContain("CritParentMissing", 2, "kubectl hns describe", nsC)
+	})
+})
+
+// Install HNC and kubectl plugin with v1alpha1.
+func setupV1alpha1(hncVersion string){
+	goPath := os.Getenv("GOPATH")
+	MustRun("kubectl apply -f https://github.com/kubernetes-sigs/multi-tenancy/releases/download/hnc-"+hncVersion+"/hnc-manager.yaml")
+	MustRun("curl -L https://github.com/kubernetes-sigs/multi-tenancy/releases/download/hnc-"+hncVersion+"/kubectl-hns -o "+goPath+"/bin/kubectl-hns")
+	MustRun("chmod +x "+goPath+"/bin/kubectl-hns")
```

Comment:
A few things here:
* I don't think there's any need to install the plugin in the gopath. You could, say, install it in `/tmp` and just change any calls to `kubectl hns` to `/tmp/kubectl-hns` and I _think_ it would all work fine.
* More importantly: I'm not sure that this is getting us enough extra coverage to be worth the effort. These are conversion tests, not e2e tests - I _think_ it's sufficient to just use checked-in v1alpha1 yaml files, which are more explicit than using the plugin anyway. That would simplify a bunch of these tests, especially the setup.

Reply:
sgtm, I will remove the verification for the v1alpha1 kubectl-hns plugin but only keep the v1alpha2 plugin test.